bin/generate_label.py

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+#
+# Copyright © 2022 Tong LI <tongli.bioinfo@protonmail.com>
+#
+# Distributed under terms of the BSD-3 license.
+
+"""
+
+"""
+import os
+import fire
+import scanpy as sc
+import numpy as np
+from skimage.draw import disk
+import tifffile as tf
+import pandas as pd
+from pathlib import Path
+from process_spaceranger import spaceranger_to_h5ad
+
+
+def main(stem, ome_md, h5ad):
+    sample_id = Path(h5ad).stem
+
+    if os.path.isdir(h5ad):
+        adata = spaceranger_to_h5ad(h5ad)
+    else:
+        adata = sc.read(h5ad)
+
+    # if multipel sections per slide -------------
+
+    # sample_ids = np.unique(adata.obs[["sample"]])
+    # print(sample_ids)
+    # md = {}
+    # for i in adata.uns["spatial"]:
+    # sample_id = re.search(".*_count_\d+_(.+)_mm.*", i).group(1)
+    # md[sample_id] = adata.uns["spatial"][i]["scalefactors"]
+    # adata.obs["Y"] = adata.obsm["spatial"][:, 0]
+    # adata.obs["X"] = adata.obsm["spatial"][:, 1]
+    # ----------------------------------------------
+
+    # check if index is integer, if not reindex
+    if not adata.obs.index.is_integer():
+        adata.obs["label_id"] = adata.obs.index
+        adata.obs.index = pd.Categorical(adata.obs.index)
+        adata.obs.index = adata.obs.index.codes
```

Comment:
Thanks for raising this, I hadn't updated the condition
We at least need them to be numerical for the label image. In a sense it would also help if we made sure the index had the smallest possible integers so the label image we write is as small as possible (related to your other comment)